ca/ca.go

```diff
@@ -320,7 +320,13 @@ func (ca *certificateAuthorityImpl) IssueCertificateForPrecertificate(ctx contex
 	ca.log.AuditInfof("Signing cert: serial=[%s] regID=[%d] names=[%s] precert=[%s]",
 		serialHex, req.RegistrationID, names, hex.EncodeToString(precert.Raw))
 
-	certDER, err := issuer.Issue(issuanceReq)
+	_, issuanceToken, err := issuer.Prepare(issuanceReq)
+	if err != nil {
+		ca.log.AuditErrf("Preparing cert failed: serial=[%s] regID=[%d] names=[%s] err=[%v]",
+			serialHex, req.RegistrationID, names, err)
+		return nil, berrors.InternalServerError("failed to sign precertificate: %s", err)
```

Comment:
I think this is a copy/paste error. This function handles signing the final certificate, not a precertificate. With that note, I think there's a logging error down on line 334 too (which was untouched by this PR).